src/app/App.js

```diff
@@ -5,20 +5,40 @@ import FilmPage from '../components/FilmPage';
 import HomePage from '../components/HomePage';
 import NotFoundPage from '../components/NotFoundPage';
 import Footer from '../components/Footer';
+import { connect } from 'react-redux';
+import { loadData } from './connect-store';
 
-function App() {
-    return (
-        <BrowserRouter>
-            <div className={styles.App}>
-                <Switch>
-                    <Route path={'/'} component={HomePage} exact />
-                    <Route path={'/film/:id'} component={FilmPage} exact />
-                    <Route path={'*'} component={NotFoundPage} />
-                </Switch>
-                <Footer />
-            </div>
-        </BrowserRouter>
-    );
+class App extends React.Component {
+    componentDidMount() {
+        this.props.loadData();
+    }
+
+    render() {
+        return this.props.genres ? (
+            <BrowserRouter>
+                <div className={styles.App}>
+                    <Switch>
+                        <Route path={'/'} component={HomePage} exact />
+                        <Route path={'/film/:id'} component={FilmPage} exact />
+                        <Route path={'*'} component={NotFoundPage} />
+                    </Switch>
+                    <Footer />
+                </div>
+            </BrowserRouter>
+        ) : (
+            <h1>Loading</h1>
```

Comment:
Please don't use such loader. Also, you can load ganres without blocking other functionality

Reply:
I write in MR Description, that we need to create task "Create LoadingComponent". I should have added something instead of downloading. And yes, we need to load genres first, and then work with the rest of the functionality. Otherwise, everything will break.

Reply:
Fixed